### ISSUE TITLE
Upgrade TF2 version and resolve Keras bug

### ISSUE THREAD

**Issue by ttbrunner** [ttbrunner/alphasheep#1]
There is a bug with the way we use Keras / the Adam Optimizer. 
For any TF version >=2.0.0rc0, training crashes because Keras tries to get gradients in eager mode.
For this reason, the TF version is currently fixed to >2.0.0b;<2.0.0rc0 in requirements.txt.

See https://github.com/tensorflow/tensorflow/issues/32646

Maybe we can use this import workaround. (Is it a workaround? Or did we call it in the wrong way?)

**Comment by ttbrunner** [comment]
We called it in the wrong way. Note: never import tf.python.keras, but always tf.keras. See https://github.com/tensorflow/tensorflow/issues/33075#issuecomment-539070546. Now training works with the current TF version (2.0.0).

This is quite the story. After changing the imports, everything works but gets really slow. Background: our "wrong" way of importing Keras made it use TF1 behaviour in predict, which is fast. Now we get TF2 behaviour - slow! See https://github.com/tensorflow/tensorflow/issues/33340, and https://stackoverflow.com/questions/58378374/why-does-keras-model-predict-slower-after-compile/58385156#58385156.

On each model.predict() call, the graph is now dynamically rebuilt. This seems to be great for large or distributed models, but introduces a huge overhead for small models.

Solutions:
1. Disable eager execution: don't want to do this globally if we don't have to
2. Don't run compile() on the model: OK for inference, but we need it for training
3. Manually write the training code in TF. I'd like to do this in the future, but no time right now. Will do when implementing other RL agents.
4. Pass `experimental_run_tf_function=False` to model.compile(). This reverts predict() behavior to the old (fast) v1 execution path - not globally, and just for this model. Also, it is not even mentioned in the docstring. Way to go - TF documentation is still a mess.

We're going with (4), which seems to be the least intrusive option.

**Comment by ttbrunner** [comment]
Fixed, TF2 release versions are used, imports are correct, and performance is back as it was.
